### DISCUSSION TITLE
Actualisation widget lors du click sur la double flêche

### DISCUSSION THREAD

**Discussion by elmerdu69** [OUARZA/Propluvia#46] [Ideas]
Hello @OUARZA, je suis de retour !

Je fais quelques essais avant de chercher à modifier le code.
- Quand je suis sur le dashboard et que je clique sur la double flèche, les valeurs du widget sont bien rafraichies, tout du moins l'heure et date de rafraichissent des données.
- Cependant quand je rafraichi ensuite la page, le widget réapparait à une taille qui n'est pas adaptée. Je dois le redimensionner à chaque fois
![image](https://github.com/OUARZA/Propluvia/assets/115868585/a309b447-3f35-4531-9e8b-e9bde8b3f91a)
- de même à chaque fois que les données sont rafraichies quotidiennement par le cron, je dois redimensionner le widget.

Est ce que tu observes le même comportement de ton coté?

**Comment by OUARZA** [comment]
Hello,
Bon retour.
Je te le confirme.
C'est pas le fait de l'avoir mis dans la case "refresh" dans le code ?
Tu sais dans ces lignes là  
// Fonction exécutée automatiquement après la sauvegarde (création ou mise à jour) de l'équipement
  public function postSave()

Mathieu

**Reply by elmerdu69** [reply]
Hello,
J'ai trouvé la raison du redimensionnent systématique du widget ! ça vient d'un bout de code que je pense avoir copié collé d'un autre plugin sans en comprendre le sens. Maintenant j'ai compris, et ça pourrait nous être utile pour fixer une taille standard du widget plus tard
```
  public function preSave() {
    $this->setDisplay("width","670px");
    $this->setDisplay("height","340px");
```
Le "refresh" dans la fonction postSave() sert justement à créer la commande manuelle "Rafraichir"


Suite à ça, j'ai tester le rafraichissent automatiques des données lorsque l'on clique sur sauvegarder sur la page de l'équipement. ça va charger les données mais continue à mouliner et finit sur une erreur de proxy. Si tu veux tester, ajoute ce bout de code dans la fonction postUpdate()
```
  public function postUpdate() {
    $cmd = $this->getCmd(null, 'refresh'); //On recherche la commande refresh de l’équipement
    if (is_object($cmd)) { //elle existe et on lance la commande
      $cmd->execCmd();
  	}
  }
```
Honnêtement je ne sais pas comment le résoudre et je propose de ne pas l'implémenter. Ce n'est pas rédhibitoire et l'on pourra toujours ajouter dans la doc que suite à la création de l’équipement pour avoir les infos la première fois, on peut soit cliquer sur la commande rafraichir de la page des commandes, ou bien la double flèche du widget. Qu'en penses-tu?

Et pour finir, j'ai trouvé un nouveau bug : quand tu as au moins 2 équipements propluvia sur la même page de dashboard, il y a un problème d'affichage du lien de l'arrêté. 
Ça vient du script JS qui n'est pas exécuté correctement par le navigateur. Je me le note à traiter mais à mon avis ça n'a pas du gêner grand monde jusqu'à maintenant.

**Comment by OUARZA** [comment]
Hey !
Cela me va de ne pas implémenter l'actualisation après la sauvegarde et d'expliquer dans la doc.
Concernant le bug, je n'ai rien vu ;)
